src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -63,186 +66,243 @@ bool FlightTaskAutoFollowTarget::activate(const vehicle_local_position_setpoint_
 {
 	bool ret = FlightTask::activate(last_setpoint);
 
-	_position_setpoint = _position;
-
 	if (!PX4_ISFINITE(_position_setpoint(0)) || !PX4_ISFINITE(_position_setpoint(1))
 	    || !PX4_ISFINITE(_position_setpoint(2))) {
 		_position_setpoint = _position;
 	}
 
-	_target_position_filter.reset(Vector3f{NAN, NAN, NAN});
-	_offset_vector_filter.reset(Vector2f(0, 0));
-	_follow_angle_filter.reset(0.0f);
-	_velocity_ff_scale.reset(0.0f);
+	_target_pose_filter.reset(Vector3f{NAN, NAN, NAN});
+	_target_pose_filter.setParameters(TARGET_POSE_FILTER_NATURAL_FREQUENCY, TARGET_POSE_FILTER_DAMPING_RATIO);
 
-	// Initialize to something such that the drone at least points at the target, even if it's the wrong angle for the perspective.
-	// The drone will move into position as soon as the target starts moving and its heading becomes known.
-	Vector2f current_drone_heading_2d{cosf(_yaw), -sinf(_yaw)};
+	// We don't command the yawspeed, since only the yaw can be calculated off of the target
+	_yawspeed_setpoint = 0.f;
+	_yaw_setpoint_filter.reset(NAN);
 
-	if (PX4_ISFINITE(current_drone_heading_2d(0)) && PX4_ISFINITE(current_drone_heading_2d(1))) {
-		_offset_vector_filter.reset(current_drone_heading_2d);
+	// Update the internally tracaked Follow Target characteristics
+	_follow_angle_rad = math::radians(get_follow_me_angle_setting(_param_nav_ft_fs.get()));
+	_follow_target_distance = _param_nav_ft_dst.get();
+	_follow_target_height = _param_nav_ft_ht.get();
 
-	} else {
-		_offset_vector_filter.reset(Vector2f(1.0f, 0.0f));
+	// Save the home position z value, to allow user so that altitude setpoint will be relative to arming position (home) altitude
+	if (_sub_home_position.get().valid_alt) {
```

Comment:
I will make it so that if there's no valid home position altitude, the mode can't be entered :thinking: . But is there a case where home position doensn't have the valid altitude, is the question.

Reply:
> But is there a case where home position doensn't have the valid altitude, is the question

not sure - @potaito ? anyway i don't like leaving things ambiguous or reliant on knowledge of other system components if possible.

Reply:
I'm sure you can find corner cases by taking off in manual without a GPS lock and then acquiring a GPS lock mid-flight for example. No home position for you then. So yes, I would assume that the drone can be flying and enter this flight mode without a valid home position.

It also makes sense to restrict the flight mode to be only allowed with a valid home position. There is a parameter for the tracking mode, and depending on its setting, the home position is required. The parameter can be changed mid-flight and during follow-me. This means there are two options:
1. You allow entering the follow-me flight mode in any case, but fall out of it as soon as home position is not valid and the parameter configuration requires it. 
2. You check the home position validity on flight task activation and reject if there is no home position.

I think the second option is more in line with where the PX4 architecture is headed.

Reply:
I was confused at first regarding what 'home position' actually is, and I guess it does get set as a local position from EKF2 when initialized.

https://github.com/PX4/PX4-Autopilot/blob/dab7b007de17b3449b875735d337149716af80c4/src/modules/commander/Commander.cpp#L1776-L1795

So then the altitude itself can actually be trusted, but it's just that it would shift a bit during flight. So I think it's okay to let user enter the mode without home position valid alt? How much can the altitude actually shift after GPS lock?

But my big questions is why would it even shift? (the local position coordinate system), or is the coordinate system being static and only the home position coordinates are changing? I find this 'home position local position coordinates affected by GPS lock' confusing 😢

Reply:
So the home position's xyz will always be valid, since it's from when the drone arms. But, the global lat, lon and alt can be incorrect. So I have added a condition to not let follow me get activated if there's no valid alt (global altitude)